core/src/main/scala/com/databricks/labs/remorph/parsers/FunctionBuilder.scala

```diff
@@ -206,33 +221,45 @@ object FunctionBuilder {
     case "UPPER" => Some(FixedArity(1))
     case "USER_ID" => Some(VariableArity(0, 1))
     case "USER_NAME" => Some(VariableArity(0, 1))
+    case "VALUE" => Some(FixedArity(2, XmlFunction))
     case "XACT_STATE" => Some(FixedArity(0))
     case "YEAR" => Some(FixedArity(1))
     case _ => None
   }
   // GCOVR_EXCL_STOP
 
+  def functionType(name: String): FunctionType = {
+    val uName = name.toUpperCase(Locale.getDefault())
+    val defnOption = functionArity(uName)
+    defnOption match {
+      case Some(fixedArity: FixedArity) => fixedArity.functionType
+      case Some(variableArity: VariableArity) => variableArity.functionType
+      case None => StandardFunction
+    }
+  }
+
   def buildFunction(name: String, args: Seq[ir.Expression]): ir.Expression = {
     val uName = name.toUpperCase(Locale.getDefault())
     val defnOption = functionArity(uName)
 
     defnOption match {
-      case Some(FixedArity(_, false)) | Some(VariableArity(_, _, false)) =>
-        // Should raise a warning/lint error here
+      case Some(fixedArity: FixedArity) if !fixedArity.isConvertible =>
         ir.UnresolvedFunction(name, args, is_distinct = false, is_user_defined_function = false)
 
-      case Some(FixedArity(arity, true)) if args.length == arity =>
+      case Some(fixedArity: FixedArity) if args.length == fixedArity.arity =>
```

Comment:
```suggestion
      case Some(FixedArity(arity, _, _)) if args.length == arity =>
```

Reply:
I woudl prefer to err in favor of the style guide here.